packages/react/src/file-upload/utils.test.tsx

```diff
@@ -168,3 +201,85 @@ describe('formatFileExtension', () => {
 		expect(formatFileExtension('.docx')).toEqual('docx');
 	});
 });
+
+describe('applyTooManyFilesError', () => {
+	it('Adds the TOO_MANY_FILES_ERROR to an existing rejection', () => {
+		const file = createExampleFile();
+
+		const rejection: FileRejection = {
+			file,
+			errors: [FILE_TOO_LARGE_ERROR_EXAMPLE],
+		};
+
+		expect(JSON.stringify(applyTooManyFilesError(rejection))).toBe(
+			JSON.stringify({
+				file,
+				errors: [FILE_TOO_LARGE_ERROR_EXAMPLE, TOO_MANY_FILES_ERROR],
+			})
+		);
+	});
+});
+
+describe('convertFileToTooManyFilesRejection', () => {
+	test('converts the file into a too many files rejection error', () => {
+		const file = createExampleFile();
+		const rejection = convertFileToTooManyFilesRejection(file);
+		expect(JSON.stringify(rejection)).toBe(
+			JSON.stringify({
+				file,
+				errors: [TOO_MANY_FILES_ERROR],
+			})
+		);
+	});
+});
+
+describe('checkRejectionsHaveSameFiles', () => {
+	const createRejectedFile = (args: Partial<FileWithStatus> = {}) =>
+		convertFileToTooManyFilesRejection(createExampleFile(args));
+
+	const fileA = createRejectedFile({ name: 'A' });
+	const fileB = createRejectedFile({ name: 'B' });
+	const fileC = createRejectedFile({ name: 'C' });
+
+	const fileD = createRejectedFile({ name: 'D' });
+	const fileE = createRejectedFile({ name: 'E' });
+	const fileF = createRejectedFile({ name: 'F' });
+
+	const rejectionListOne = [fileA, fileB, fileC];
+	const rejectionListTwo = [fileD, fileE, fileF];
+	const rejectionListThree = [
+		{
+			...fileA,
+			errors: [{ code: ErrorCode.FileTooLarge, message: 'too large' }],
+		},
+		fileB,
+		fileC,
+	];
+
+	test('exactly same', () => {
+		expect(
+			checkRejectionsHaveSameFiles(rejectionListOne, rejectionListOne)
+		).toBe(true);
+	});
+	test('different', () => {
+		console.log(JSON.stringify({ rejectionListOne }));
+		console.log('nn', JSON.stringify({ rejectionListTwo }), 'nn');
+		expect(
+			checkRejectionsHaveSameFiles(rejectionListOne, rejectionListTwo)
+		).toBe(false);
+	});
+	test('same files but different errors', () => {
+		expect(
+			checkRejectionsHaveSameFiles(rejectionListOne, rejectionListThree)
+		).toBe(true);
+	});
+});
+
+test('removeItemAtIndex', () => {
+	const items = ['a', 'b', 'c'];
+	expect(JSON.stringify(removeItemAtIndex(items, 1))).toBe(
```

Comment:
You shouldn't need to stringify this one. This should be enough:

```ts
expect(removeItemAtIndex(items, 1)).toEqual(['a', 'c'])
```

Reply:
Implemented your suggestion